server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailConverter.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.mailet.AttributeValue;
+import org.apache.mailet.Mail;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+class MailConverter {
+    static final AttributeName ORIGIN_MAILBOXES_ATTRIBUTE_NAME = AttributeName.of("originMailboxes");
+    static final AttributeName HAS_ATTACHMENT_ATTRIBUTE_NAME = AttributeName.of("hasAttachment");
+    static final AttributeName OWNER_ATTRIBUTE_NAME = AttributeName.of("owner");
+    static final AttributeName DELIVERY_DATE_ATTRIBUTE_NAME = AttributeName.of("deliveryDate");
+    static final AttributeName DELETION_DATE_ATTRIBUTE_NAME = AttributeName.of("deletionDate");
+    static final AttributeName SUBJECT_ATTRIBUTE_VALUE = AttributeName.of("subject");
+
+    private final MailboxId.Factory mailboxIdFactory;
+    private final MessageId.Factory messageIdFactory;
+
+    MailConverter(MailboxId.Factory mailboxIdFactory, MessageId.Factory messageIdFactory) {
+        this.mailboxIdFactory = mailboxIdFactory;
+        this.messageIdFactory = messageIdFactory;
+    }
+
+    Mail toMail(DeletedMessage deletedMessage) throws MessagingException {
+        return MailImpl.builder()
+            .name(deletedMessage.getMessageId().serialize())
+            .sender(deletedMessage.getSender())
+            .recipients(deletedMessage.getRecipients())
+            .mimeMessage(new MimeMessage(Session.getDefaultInstance(new Properties()), deletedMessage.getContent().get()))
+            .attribute(OWNER_ATTRIBUTE_NAME.withValue(SerializableUser.toAttributeValue(deletedMessage.getOwner())))
+            .attribute(HAS_ATTACHMENT_ATTRIBUTE_NAME.withValue(AttributeValue.of(deletedMessage.hasAttachment())))
+            .attribute(DELIVERY_DATE_ATTRIBUTE_NAME.withValue(SerializableDate.toAttributeValue(deletedMessage.getDeliveryDate())))
+            .attribute(DELETION_DATE_ATTRIBUTE_NAME.withValue(SerializableDate.toAttributeValue(deletedMessage.getDeletionDate())))
+            .attribute(ORIGIN_MAILBOXES_ATTRIBUTE_NAME.withValue(AttributeValue.of(serializedMailboxIds(deletedMessage))))
+            .attribute(SUBJECT_ATTRIBUTE_VALUE.withValue(subjectAttributeValue(deletedMessage)))
+            .build();
+    }
+
+    DeletedMessage fromMail(Mail mail) {
+        MemoizedSupplier<byte[]> byteContent = new MemoizedSupplier<>(Throwing.supplier(
+            () -> MimeMessageUtil.asBytes(mail.getMessage())));
+
+        return DeletedMessage.builder()
+            .messageId(messageIdFactory.fromString(mail.getName()))
+            .originMailboxes(retrieveMailboxIds(mail))
+            .user(retrieveOwner(mail))
+            .deliveryDate(retrieveDate(mail, DELIVERY_DATE_ATTRIBUTE_NAME))
+            .deletionDate(retrieveDate(mail, DELETION_DATE_ATTRIBUTE_NAME))
+            .sender(mail.getMaybeSender())
+            .recipients(mail.getRecipients())
+            .content(() -> new ByteArrayInputStream(byteContent.get()))
```

Comment:
see my comment in previous PR: you probably recreated MimeMessageInputStreamSource

Reply:
Fixed in https://github.com/linagora/james-project/pull/2194/commits/36135ebf5ad5f2e5f108f368b2861f22f3d91ba5